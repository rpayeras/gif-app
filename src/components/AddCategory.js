import React, { useState } from 'react';
import PropTypes from 'prop-types'

export const AddCategory = ({onSubmit}) => {
    const [inputText, setInputText] = useState('')

    const handleInput = (e) => setInputText(e.target.value);

    const handleSubmit = (e) => {
        e.preventDefault()

        if(inputText.trim().length > 2){
            onSubmit(inputText)
            setInputText('')
        }
    }

    return (
        <>
            <form onSubmit={handleSubmit}>
                <input name="name" type="text" value={inputText} onChange={handleInput} placeholder="Add a category..."/>
            </form>
        </>
    );
};

AddCategory.propTypes = {
    onSubmit: PropTypes.func.isRequired
}