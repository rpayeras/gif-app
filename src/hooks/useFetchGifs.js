import { useState, useEffect } from "react"
import { getGifs } from '../helpers/getGifs';
  // const [images, setImages] = useState([]);

export const useFetchGifs = (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true,
    });

    useEffect(() => {
        getGifs(category)
        .then(imgs => setState({
            data: imgs,
            loading: false
        }))
    }, [category])

    return state;
}