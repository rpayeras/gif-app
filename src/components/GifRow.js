import React, { useState, useEffect } from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifRow = ({category}) => {
  const {data: images, loading} = useFetchGifs(category)

  return (
    <>
      <h3>{category}</h3>
      {loading ?? <p>'Loading...'</p>}
      { <div className='card-grid '>
        {images.map((img) => {
          return (
            <GifGridItem {...img} key={img.id} />
          )
        })}
    </div> }
    </>
  );
};

GifRow.propTypes = {
  category: PropTypes.string.isRequired
}