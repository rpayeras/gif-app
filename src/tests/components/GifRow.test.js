import React from 'react';

import { shallow } from 'enzyme';
import { GifRow } from '../../components/GifRow';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('Testing GifGrid component', () => {
    test('it should renders properly', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        })
        const wrapper = shallow(<GifRow category='test'/>)
        expect(wrapper).toMatchSnapshot()
    })

    test('should show items using useFetchGifs', () =>{
        const gifs = [{
            id: 'ABC',
            url: 'https://testing.com/test.jpg',
            title: 'test'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true,
        })

        const wrapper = shallow(<GifRow category='test'/>)

        //expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    })
})