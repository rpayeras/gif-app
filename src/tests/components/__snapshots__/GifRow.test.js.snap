// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGrid component it should renders properly 1`] = `
<Fragment>
  <h3>
    test
  </h3>
  <Component />
  <div
    className="card-grid "
  />
</Fragment>
`;
